src/basicgate.jl

```diff
@@ -0,0 +1,10 @@
+struct BasicGate{SYM} end
+for (SYM, OP) in [:∨ => (|), :∧ => (&), :⊻ => (⊻)]
+    @eval nin(::BasicGate{$(QuoteNode(SYM))}) = 2
+    @eval function (bg::BasicGate{$(QuoteNode(SYM))})(p, q)
+        return $OP(p, q)
+    end
+end
+nin(::BasicGate{:¬}) = 1
+(bg::BasicGate{:¬})(p::T) where T = T(!Bool(p))
```

Comment:
To convince you that abstraction is good. I implemented the gate interfaces on the BasicGate type. The gadget searching directly works.

@Rose-max111

Reply:
I see. That's a good solution for general truth table mapping